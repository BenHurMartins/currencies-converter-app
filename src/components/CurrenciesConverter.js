import { useState } from "react";
import '../styles.css';
import Selector from "./Selector";
import { convertCurrency } from "../api";

const CurrenciesConverter = (props) => {

    const [fromCurrency, setFromCurrency] = useState('USD')
    const [toCurrency, setToCurrency] = useState('EUR')
    const [fromValue, setFromValue] = useState(0)
    const [toValue, setToValue] = useState(0)

    const handleFromValueChange = (e) => {
        const value = e.target.value.replace(',', '.')
        if (!isNaN(value)) {
            setFromValue(value)
        }
    }
    const handleFromCurrencyChange = (e) => {
        setFromValue(0)
        setFromCurrency(e.target.value);
    }
    const handleToCurrencyChange = (e) => {
        setToValue(0)
        setToCurrency(e.target.value);
    }

    const handleConversion = () => {
        const value = parseFloat(fromValue).toFixed(2) == 0 ? parseFloat(1).toFixed(2) : parseFloat(fromValue).toFixed(2)
        setFromValue(value)

        convertCurrency(fromCurrency, toCurrency, value)
            .then(result => {
                setToValue(result)
            })
            .catch(err => {
                alert("Error trying to convert currencies, try again later.")
            })
            .catch(err => {
                alert("Error: " + err.error_message)
            })
    }

    return (
        <>
            <p className="app-subtitle">
                Currency
            </p>
            <div className="main-container">
                <div className="form-container">
                    <div className="half-container">
                        <p className="label">From:</p>
                        <input placeholder="0.00" min="0" value={fromValue} step="0.01" title="Currency" pattern="^\d+(?:\.\d{1,2})?$" onChange={handleFromValueChange} />
                        <Selector value={fromCurrency} onChange={handleFromCurrencyChange} options={props.currenciesArray} currencies={props.currencies} />
                    </div>
                    <div className="half-container">
                        <p className="label">To:</p>
                        <input placeholder="0.00" value={toValue} />
                        <Selector value={toCurrency} onChange={handleToCurrencyChange} options={props.currenciesArray} currencies={props.currencies} />
                    </div>
                </div>
                <div>
                    <button onClick={handleConversion}>Convert</button>
                </div>
            </div>
        </>
    )
}

export default CurrenciesConverter;