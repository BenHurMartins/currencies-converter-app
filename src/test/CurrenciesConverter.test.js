import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import CurrenciesConverter from "../components/CurrenciesConverter";
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });
    
describe('CurrencyConverter', () => {

    it('should render the component', ()=> {
        const optionsArray=[
            'USD', 'BRL'
        ]
        const currencies = {
            'USD': 'United States Dolar',
            'BRL': 'Brazilian Real'
        }
        const component = renderer.create(<CurrenciesConverter lang={'EN'} currenciesArray={optionsArray} currencies={currencies}/>)
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('should show the text in english', () => {
        const cryptoInstance = shallow(<CurrenciesConverter lang={'EN'} />);
        const element = cryptoInstance.find('.app-subtitle');
        expect(element.text()).toBe('Currency');
    }); 
    
    it('should be in portuguese', () => {
        const cryptoInstance = shallow(<CurrenciesConverter lang={'PT_BR'} />);
        const element = cryptoInstance.find('.app-subtitle');
        expect(element.text()).toBe('Moeda');
    })
})
