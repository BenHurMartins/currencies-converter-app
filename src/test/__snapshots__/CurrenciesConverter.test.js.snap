// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyConverter should render the component 1`] = `
Array [
  <p
    className="app-subtitle"
  >
    Currency
  </p>,
  <div
    className="main-container"
  >
    <div
      className="form-container"
    >
      <div
        className="half-container"
      >
        <p
          className="label"
        >
          From:
        </p>
        <input
          min="0"
          onChange={[Function]}
          pattern="^\\\\d+(?:\\\\.\\\\d{1,2})?$"
          placeholder="0.00"
          step="0.01"
          title="Currency"
          value={0}
        />
        <select
          onChange={[Function]}
          value="USD"
        >
          <option
            value="USD"
          >
            United States Dolar
          </option>
          <option
            value="BRL"
          >
            Brazilian Real
          </option>
        </select>
      </div>
      <div
        className="half-container"
      >
        <p
          className="label"
        >
          To:
        </p>
        <input
          placeholder="0.00"
          value={0}
        />
        <select
          onChange={[Function]}
          value="EUR"
        >
          <option
            value="USD"
          >
            United States Dolar
          </option>
          <option
            value="BRL"
          >
            Brazilian Real
          </option>
        </select>
      </div>
    </div>
    <div>
      <button
        onClick={[Function]}
      >
        Convert
      </button>
    </div>
  </div>,
]
`;
